### DISCUSSION TITLE
Github Desktop cannot find local git folder

### DISCUSSION THREAD

**Discussion by Techmokid** [community/community#24077] [Apps, API and Webhooks]
<p>Hello!</p>
<p>I seem to have an issue regarding the github desktop. I’ve had no issues whatsoever with it up until yesterday. It seems to work perfectly fine on my system, up until I try to access my repo at <a href="https://github.com/Techmokid/My-Projects" rel="noopener nofollow ugc">https://github.com/Techmokid/My-Projects</a>. Lately I’ve been trying to finish off an AI, but the actual saved network is quite large. I don’t just mean physical disk space (About 8.94 GB, I am aware that I’m close to getting an email from github if I commit that), but also the sheer number of files created (1.5 million files). I expect it to take awhile, but I didn’t expect GHD to just cease being able to see the folder entirely. When I try to manually search for the folder, it’s already looking where all my projects are. But when I hit “Select folder”, after a few seconds of thinking with “Fetching origin”, it just reverts back to “Cannot find ‘my-projects’”.</p>
<p>Is there a limit to the number of files I can have in a repo before github desktop just decides it cannot see the folder anymore? Whether my github desktop can see the local clone of the repo or not seems to correlate with whether or not I put in that folder (It’s just a whole ton of .txt files). And if there is a limit to github desktop, how would I go about commiting this file to my github repo (Do I have to learn the CLI, or am I just missing a setting somewhere)?</p>
<p>Again, it has been working fine up until yesterday (Been working on this for quite a few months now). I don’t understand what I’m missing here. Earlier it decided to force me to “Publish new branch” (Then after awhile it decided I didn’t have to), despite me not telling it to, and now all this.</p>
<p>I would prefer to keep my repo private if that’s possible because it contains some important projects of mine.</p>
<p>If anyone could at least give me some pointers or something, that would be amazing!</p>
<p>Kind regards,<br>
Andrey</p>

**Comment by steveward** [comment, marked as answer]
<p>Welcome <a class="mention" href="/techmokid">@Techmokid</a>!</p>
<p>I believe what you are running into is the diff limit in GitHub Desktop. We set this limit to 20MB, and your repository is likely over that if you are diffing 1.5million files.</p>
<p>As a workaround, if you make the first commit using the Git command line and then drag-and-drop the repository into the application again, that should get you back to a good place. I would also recommend looking into using Git LFS if you are working with large files.</p>
<p>Let me know if this helps or if you have any further questions.</p>

**Comment by Techmokid** [comment]
<p>It was definitely the diff limit. I solved the issue in 2 ways. First, I figured out that I can change things folder by folder, and as long as I’m careful about how MANY I change at once, it works. The other alternative is to keep the multimillion file containing folder outside of my github repo entirely, avoiding the need for the issue to show up in the first place. Thankyou!</p>
